specification/cognitiveservices/data-plane/EntitySearch/v1.0/EntitySearch.json

```diff
@@ -0,0 +1,935 @@
+              "$ref": "#/definitions/EntitiesEntityPresentationInfo"
+            },
+            "bingId": {
+              "type": "string",
+              "description": "An ID that uniquely identifies this item."
+            }
+          }
+        }
+      ]
+    },
+    "ContractualRulesContractualRule": {
+      "x-ms-discriminator-value": "ContractualRules/ContractualRule",
+      "discriminator": "_type",
+      "type": "object",
+      "required": [
+        "_type"
+      ],
+      "properties": {
+        "targetPropertyName": {
+          "description": "The name of the field that the rule applies to.",
+          "type": "string"
+        },
+        "_type": {
+          "type": "string"
+        }
+      }
+    },
+    "ImageObject": {
+      "description": "Defines an image",
+      "allOf": [
+        {
+          "$ref": "#/definitions/MediaObject"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "thumbnail": {
+              "description": "The URL to a thumbnail of the image",
+              "$ref": "#/definitions/ImageObject"
+            }
+          }
+        }
+      ]
+    },
+    "Place": {
+      "description": "Defines information about a local entity, such as a restaurant or hotel.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Thing"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "address": {
+              "description": "The postal address of where the entity is located",
+              "$ref": "#/definitions/PostalAddress"
+            },
+            "telephone": {
+              "description": "The entity's telephone number",
+              "type": "string"
+            }
+          }
+        }
+      ]
+    },
+    "Organization": {
+      "description": "Defines an organization.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Thing"
+        },
+        {
+          "type": "object",
+          "properties": {}
+        }
+      ]
+    },
+    "ResponseBase": {
+      "discriminator": "_type",
+      "type": "object",
+      "required": [
+        "_type"
+      ],
+      "properties": {
+        "_type": {
+          "type": "string"
+        }
+      }
+    },
+    "Error": {
+      "description": "Defines the error that occurred.",
+      "type": "object",
+      "required": [
+        "Code",
+        "Message"
+      ],
+      "properties": {
+        "code": {
+          "description": "The error code that identifies the category of error.",
+          "type": "string",
+          "default": "None",
+          "enum": [
+            "None",
+            "ServerError",
+            "InvalidRequest",
+            "RateLimitExceeded",
+            "InvalidAuthorization",
+            "InsufficientAuthorization"
+          ],
+          "x-ms-enum": {
+            "name": "ErrorCode",
+            "modelAsString": false
```

Comment:
I consider actually modeling this as a string: If your API ever returns a code that's not in the `enum` above, deserialization in generated SDKs/CLIs will fail... failing to deserialize an error response is a bad error to have 😄

Reply:
All of our definitions are auto generated from our models. We don't really want to special case one enum over others, but funny enough this isn't even the enum that we expect to change frequently.. there are others that have higher modification rates (not present in this particular search API).

You're saying if the deserializer tries to deserialize an enum value that it doesn't know about, it'll throw an exception instead of returning the default of that type (I believe basically all our enum fields are optional, so I would hope that would be a null)?? That seems funky and also probably means we literally can't rely on enums at all right?

Modelling enums as strings basically loses the semantic value of having a limited set of values, doesn't it? Void pointer shudders. Does autorest end up generating a string property or some kind of whacky string-backed enum?

Reply:
Defer to you on this, we definitely have enums that we know will have additional values added over the lifecycle of the SDK. If the autorest generated code will throw when trying to deserialize an unknown value, we'll have to model them all as strings. Otherwise it would be awesome if the deserializer could set the enum / enum? to its default type value (which I think for c# enums would be the first value?)

Reply:
The problem with defaulting to something is that we *have* to generate code that is able to round-trip objects without mutation. I.e. if somebody does a GET, modifies one property and then does a PUT/POST, you don't wanna have any of the other properties changed unexpectedly - an exception is better in that case (is what was once decided)!

But: The real issue is the fact we rely on C# `enum`s too much, we should rather generate something as propesed here: https://github.com/Azure/autorest/issues/1591
This is high (if not the highest) on our list for the next features to implement, so it somewhat depends on your release plans. The "new" way will be compatible with both the `enum` way (stuff like `Color color = Color.Red` will work the same way) and the string way (there will likely be implicit conversions to/from `string`), so existing code will keep working unless fancy things are done. So if you release soon (before we get to do the above - I do not promise any timeline), you may choose whichever you think is best for customers (and I get that `enum`s have the huge benefit of being auto-discoverable and you don't have to *know* that there's also a static class full of string constants...)

Then, when you rerelease the SDK in the future:
- in the unlikely event that you didn't change any enums: congrats, don't do anything, otherwise
- you have a breaking change, but then you can choose whether you wanna move to the new way of generating enums that will cause less pain in the long run

Reply:
Huh, extensible enums. Sounds pretty neat. Thanks for the explanation (the round-tripping concern makes sense, our APIs are pretty simple and meant to be read-only for the most part, so it's not really something we've thought about). 

I'll opt to keep this a real enum for now, in part because this API doesn't really have enums that we expect to modify in the foreseeable future